### ISSUE TITLE
Should we lint things that would get fixed by rustfmt anyway?

### ISSUE THREAD

**Issue by oli-obk** [Manishearth/rust-clippy#793]
rustfmt would/(will?) reorder the following to move all the `use`s to the top.

```rust
use a::b;
fn bla() {}
use x::y;
```

We could of course lint on such things, but people might end up having to reconfigure both rustfmt and clippy for their projects, because the one moves something around that clippy then lints about.

**Comment by llogiq** [comment]
As long as we don't lint something that rustfmt *introduces*, we should be good. And if we did, I'd consider this either a bug in clippy or rustfmt.

**Comment by oli-obk** [comment]
It's also code-duplication if rustfmt would fix it anyway.

**Comment by llogiq** [comment]
Perhaps, but not everyone uses rustfmt (at least yet) and if linting it isn't too complex there is little harm.

**Comment by sunjay** [comment]
Just to add to this -- some people prefer being just warned about a problem through a lint rather than trusting a program to make the changes itself. Having rustfmt warnings in Rust linters can definitely be a good thing.

**Comment by felix91gr** [comment]
What's the consensus on this nowadays? It feels to me like it might've been settled by now in the Project Goals of both Clippy and `rustfmt`.

(fwiw I'm asking to know if this Issue can be closed. Some cleanup work that Clippy could use)

---

I'm more than 8 years late to the conversation, but if you ask me... I'd rather `rustfmt` have a warning mode or an API to ask for warnings, instead of Clippy having to lint for formatting. But that's me, I just don't like linters that ping me about formatting.

**Comment by hkBst** [comment]
Is there really a big category like this, or are there just a few things that can be decided on a case by case basis?
